constant_time_derive/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Data, DataEnum, DeriveInput, Fields, GenericParam, Generics};
+
+#[proc_macro_derive(ConstantTimeEq)]
+pub fn constant_time_eq(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    derive_ct_eq(&input)
+}
+
+// TODO: Check or remove padding and align decorators on the struct
+fn parse_fields(fields: &Fields) -> Result<proc_macro2::TokenStream, &'static str> {
+    match &fields {
+        Fields::Named(fields_named) => {
+            let mut token_stream = quote!();
+            let mut iter = fields_named.named.iter().peekable();
+
+            while let Some(field) = iter.next() {
+                let ident = &field.ident;
+                match iter.peek() {
+                    None => token_stream.extend(quote! { {self.#ident}.ct_eq(&{other.#ident}) }),
+                    Some(_) => {
+                        token_stream.extend(quote! { {self.#ident}.ct_eq(&{other.#ident}) & })
+                    }
+                }
+            }
+            Ok(token_stream)
```

Comment:
:clown_face: I don't know that this suggestion is much better. The while loop and peeking for "is there more" felt off
```suggestion
            let mut token_stream = quote!();
            let separators = ::core::iter::once("").chain(::core::iter::repeat("&"));
            let items = separators.zip(fields_named.named.iter());
            for (separator, field) in items {
                let ident = &field.ident;
                let separator_token: proc_macro2::TokenStream = separator.parse().expect("Should be a valid token");
                token_stream.extend(quote! { #separator_token {self.#ident}.ct_eq(&{other.#ident}) });
            }
            Ok(token_stream)
```

Reply:
I felt peeking is more readable but thanks for this, I might use this elsewhere